### PR TITLE
homogenize object cols to str

### PR DESCRIPTION
**Pull request by dcolinmorgan**
g.umap() works even after adding this column to the untitled.json data

```
data['test']=data['Source_Proto'].apply(lambda x: x[0])
data['test'][[1,20,35,72,90]]=1
data['test'][[10,5,16,28,75]]=1.0
data['test'][[12,7]]='NaN'
data['test'][[13,8]]=np.NaN
```

### CODE REVIEW

**Review by lmeyerov — APPROVED** [review]
looks good -- can you log a message that the retry is happening + on which columns?

File: graphistry/feature_utils.py [hunk 1]
```diff
@@ -907,6 +907,7 @@ def process_dirty_dataframes(
         try:
             X_enc = data_encoder.fit_transform(ndf, y)
         except TypeError:
+            logger.info("obj columns:", object_columns, "are being converted to str")
```
Comment (lmeyerov) [inline graphistry/feature_utils.py:910]: @dcolinmorgan `logger.info("obj columns: %s are being converted to str", object_columns)`

Comment (lmeyerov) [reply graphistry/feature_utils.py:910]: + ci failing?

Comment (dcolinmorgan) [reply graphistry/feature_utils.py]: message was placed before object_columns was defined

**Review by lmeyerov — APPROVED** [review]

File: graphistry/feature_utils.py [hunk 2]
```diff
@@ -904,7 +904,14 @@ def process_dirty_dataframes(
...
         
-        X_enc = data_encoder.fit_transform(ndf, y)
+        try:
+            X_enc = data_encoder.fit_transform(ndf, y)
+        except TypeError:
+            nndf = ndf.copy()
+            object_columns = nndf.select_dtypes(include=['object']).columns
+            nndf[object_columns] = nndf[object_columns].astype(str)
+            X_enc = data_encoder.fit_transform(nndf, y)
+            logger.info("obj columns: %s are being converted to str", object_columns)
```
Comment (lmeyerov) [inline graphistry/feature_utils.py:914]: can we get a test that triggers & handles this edge case?

can land after, looks good!

**Review by lmeyerov — CHANGES_REQUESTED** [review]
see comments

**Review by lmeyerov — APPROVED** [review]
